deep_quoridor/src/main.py

```diff
@@ -1,87 +1,46 @@
-    }
-
-    if render == "print":
-        print("Initial Board State:")
-        board = game.render()
-        print(board)
-    elif render == "curses":
-        stdscr = curses.initscr()
-        curses.noecho()
-        curses.cbreak()
-        curses.start_color()
-
-    step = 0
-
-    for agent in game.agent_iter():
-        _, _, termination, truncation, _ = game.last()
-        if termination or truncation:
-            action = None
-            winner = max(game.rewards, key=game.rewards.get)
-            print(f"\nGame Over! {winner} wins after {step} steps.")
-            break
-
-        action = agents[agent].get_action(game)
-        print(f"\nStep {step + 1}: {agent} takes action {action}")
-        game.step(action)  # Apply action
-        print(f"Rewards: {game.rewards}")
-
-        board = game.render()
-
-        if render == "curses":
-            stdscr.clear()
-            stdscr.addstr(2, 2, board)
-            stdscr.refresh()
-            curses.napms(100)
-
-        if render == "print":
-            print(board)
-
-        step += 1
-
-    if render == "curses":
-        curses.nocbreak()
-        curses.echo()
-        curses.endwin()
-
-    game.close()
-
+from random_agents import RandomAgent
+from renderers import ResultsRenderer, TextRenderer, CursesRenderer
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="Deep Quoridor")
     parser.add_argument("-N", "--board_size", type=int, default=None, help="Board Size")
     parser.add_argument("-W", "--max_walls", type=int, default=None, help="Max walls per player")
-    parser.add_argument("-r", "--render", choices=["print", "curses"], default="print", help="Render mode")
+    parser.add_argument("-r", "--render", choices=["result", "text", "curses"], default="result", help="Render mode")
     parser.add_argument("--step_rewards", action="store_true", default=False, help="Enable step rewards")
+    parser.add_argument(
+        "--games_output_filename",
+        type=str,
+        default="game.yaml",
+        help="Save the played games to a file. Use 'None' to disable saving.",
+    )
 
     args = parser.parse_args()
-    play(**vars(args))
+
+    # TO DO: this should be automatically detected from the environment
+    renderer = None
+    if args.render == "result":
+        renderer = ResultsRenderer()
+    elif args.render == "text":
+        renderer = TextRenderer()
+    elif args.render == "curses":
+        renderer = CursesRenderer()
+
+    saver = None
+    if args.games_output_filename != "None":
+        saver = ArenaYAMLRecorder(args.games_output_filename)
+
+    args = {
+        "board_size": args.board_size,
+        "max_walls": args.max_walls,
+        "step_rewards": args.step_rewards,
+        "renderer": renderer,
+        "saver": saver,
+    }
+
+    args = {k: v for k, v in args.items() if v is not None}
+    arena = Arena(**args)
+    arena.play_game(RandomAgent(), SimpleAgent())
```

Comment:
TO DO: this should be also be configured from the parameters, e.g. `--players=RandomAgent,SimpleAgent`

Reply:
or via a YAML file that describes a particular run with all its details. Sort of like a tournament specification.